reference/vane-apis/behn.md

```diff
@@ -0,0 +1,366 @@
++++
+title = "Behn"
+weight = 2
+template = "doc.html"
++++
+
+# Behn
+
+In this document we describe the public interface for Behn. Namely, we describe
+each `task` that Behn can be `pass`ed, and which `gift`(s) Behn can `give` in return.
+
+
+## Tasks
+
+### `%born`
+
+Each time you start your Urbit, the Arvo kernel calls the `%born` task for Behn. When
+called, Behn gets the current time from Unix and updates its list of timers
+accordingly.
+
+#### Accepts
+
+```hoon
+[~]
+```
+
+#### Returns
+
+`%born` does not return a `gift`.
+
+#### Source
+```hoon
+::  +born: urbit restarted; refresh :next-wake and store wakeup timer duct
+  ::
+  ++  born  set-unix-wake(next-wake.state ~, unix-duct.state duct)
+```
+
+
+### `%crud`
+
+`%crud` is called whenever an error involving Behn occurs, which for now means
+that Behn has no timers in its state.
```

Comment:
The most common error is actually that whoever Behn woke up crashed.  In this case, Behn gets the `%crud`, which it passes along to the person Behn tried to wake up as a `%wake` with a non-null `error`.

Reply:
I need a little more clarity. Does this sound right? Behn attempts to wake up Ames. Ames has crashed, so it passes Behn a `%crud` task, and Behn then responds by giving Ames a `%wake` gift with an error message?

I considered a few other possibilities of what you meant but this seemed like the simplest one (though maybe incorrect).

Reply:
Behn attempts to wake up Ames, but Ames crashes.  This whole event (not just move) is thrown away -- nobody can react to a crash directly.  However, vere happens to notice that the event didn't complete, so it injects a *new* event that says "you were going to try to process an event, but it failed, and here's a stack trace in case you want to handle that (eg print it out)".

Reply:
How does this look? https://github.com/urbit/docs/pull/817/commits/f268d6d8752e037364587e59656478f8177c309d

Reply:
Sure, this works.